traffic_ops/app/lib/API/DeliveryServiceRegexes.pm

```diff
@@ -66,6 +73,12 @@ sub index {
 		return $self->not_found();
 	}
 
+	my $tenant_utils = Utils::Tenant->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db();
+	if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id)) {
+		return $self->forbidden();
```

Comment:
actually, can you do that on all your forbidden message for now on that are the result of a tenancy check failure?